src/components/InteractionLayer/index.js

```diff
@@ -287,8 +438,37 @@ class InteractionLayer extends React.Component {
       );
     }
     const annotations = this.props.annotations.map(a => (
-      <Annotation key={a.id} {...a} xScale={xScale} height={height} />
+      <Annotation key={a.id} {...a} height={height} />
     ));
+    const xScale = createXScale(this.props.subDomain, width);
+    const areas = this.props.areas.map(a => {
+      let a1 = a;
```

Comment:
Should there be a spread operator?

Reply:
No, this is a bit weird. We're okay with reading the properties from `a` normally, but if we *did* create an area, then we copy it. Let me clean this up to make that a bit more obvious.